projects/telemetry/src/nmea.c

```diff
@@ -0,0 +1,33 @@
+#include "nmea.h"
+#include <stdbool.h>
+#include <string.h>
+
+static char s_hex[] = "0123456789ABCDEF";
+
+// out must be a pointer to at least 3 chars
+void prv_int_to_hex(uint8_t checksum, char *out) {
+  // we know that it'll be a 2 digit hex number since the checksum is 8-bit
+  out[0] = s_hex[checksum >> 4];
+  out[1] = s_hex[checksum & 15];
+  out[2] = '\0';
+}
+
+void nmea_compute_checksum(char *message, size_t message_len, char *out) {
+  uint8_t sum = 0;
+  for (uint8_t i = 1; message[i] != '*' && i < message_len; i++) {
+    sum ^= message[i];
+  }
+  prv_int_to_hex(sum, out);
+}
+
+bool nmea_compare_checksum(char *message, size_t message_len) {
+  char *received = message + message_len - 2;
```

Comment:
Could you explain what this is doing? I'm not sure why you're getting the pointer of a near-end value and then subtracting 1 from the pointer below before dereferencing it. It seems a little roundabout and difficult to understand.

Reply:
It might help if you had some higher level explanations of what this checksum is doing and how it fits into the message format in your header.